Automattic/wp-calypso
# Add pre-purchase notice for scan product overlaps

#### Changes proposed in this Pull Request

* This PR adds support for pre-purchase notices surrounding anti-spam overlaps.
  * If a user already has Jetpack Anti-Spam, and attempts to purchase another plan that includes Anti-Spam, they will be shown a notice.
  * If a user already has a plan with anti-spam, and attempts to directly purchase a Jetpack Anti-Spam, they will be shown a notice.
* This is achieved by implementing the same logic as the backup and anti-spam-related pre-purchase notices, except with scan product and feature slugs.


#### Testing instructions



* Set up a WordPress site with Jetpack.
* Purchase Jetpack Scan.
* Add a plan that also includes scan to the cart, such as Jetpack Security.
* Verify the notice is displayed:

<img width="585" alt="Screen Shot 2022-03-04 at 11 04 15 AM" src="https://user-images.githubusercontent.com/10933065/156817670-e937b3b4-1b56-4ef8-9467-710111d35b27.png">

* Remove your Scan product.
* Purchase a plan that includes Scan, such as Jetpack Security.
* Add Jetpack Scan to the cart.
  (You can force this by opening [http://calypso.localhost:3000/checkout/[YOUR_SITE]/jetpack_scan](http://calypso.localhost:3000/checkout/%5BYOUR_SITE%5D/jetpack_anti_scan))
* Verify the notice is displayed:

<img width="585" alt="Screen Shot 2022-03-04 at 10 52 10 AM" src="https://user-images.githubusercontent.com/10933065/156817661-8993bc26-35dd-46da-ac91-75abd11914e2.png">



Related to https://github.com/Automattic/wp-calypso/pull/61577
